src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -945,8 +945,17 @@ public void Connect(string host, int port)
                 throw new NotSupportedException(SR.net_invalidversion);
             }
 
-            IPAddress[] addresses = Dns.GetHostAddressesAsync(host).GetAwaiter().GetResult();
-            Connect(addresses, port);
+            IPAddress parsedAddress;
```

Comment:
This also doesn't hurt.  One of the first things Dns.GetHostAddresses does is this exact same thing.  So it's already being done on Windows, this just moves it up earlier.  For cases where it succeeds in parsing, it's appx twice as fast as calling Dns.GetHostAddressesAsync(...).GetAwaiter().GetResult(), and for cases where I fails in parsing, it's so small in comparison to the work done by GetHostAddressesAsync that it doesn't show up on a benchmark.  That's why I did it here rather than contorting the PAL for it.